test/patient.js

```diff
@@ -1938,6 +1939,69 @@ tap.test('patient should support standard _id parameter', (t) => {
   })
 })
 
+tap.test('patient update should insert document when upsert true and document does not exist', (t) => {
+  env.initDB((err, db) => {
+    t.error(err)
+
+    server.start((err) => {
+      t.error(err)
+
+      config.setConf('operations:upserting', true)
+
+      const pat = _.cloneDeep(require('./resources/Patient-1.json'))
+
+      request.put({
+        url: `http://localhost:3447/fhir/Patient/${pat.id}`,
+        headers: headers,
+        body: pat,
+        json: true
+      }, (err, res, body) => {
+        t.error(err)
+        t.equal(res.statusCode, 201, 'response status code should be 201')
+        t.ok(body)
+
+        env.clearDB((err) => {
+          t.error(err)
+          server.stop(() => {
+            config.setConf('operations:upserting', false)
+            t.end()
+          })
+        })
+      })
+    })
+  })
+})
+
+tap.test('patient update should error when upsert false and document does not exist', (t) => {
+  env.initDB((err, db) => {
+    t.error(err)
+
+    server.start((err) => {
+      t.error(err)
+
```

Comment:
Lets explicitly set the config value to false so that its indicated its set to false. This is dependant on the default value in the config file which can be changed without chaining anything in the test